modelina-website/src/store/useLayoutStore.tsx

```diff
@@ -0,0 +1,153 @@
+import type { Draft, Immutable } from 'immer';
```

Comment:
Why do we need this?

Reply:
It's one of react design's core principles which is [immutability in state](https://react.dev/learn/updating-objects-in-state), sometimes when the state is not immutable it also doesn't trigger the re-rendering process, the state is updated but not changed on the UI

Reply:
Also, object reference uses heap allocation when the object property value changes, it's still using the same memory address